src/tls.c

```diff
@@ -786,6 +787,43 @@ static int connTLSWrite(connection *conn_, const void *data, size_t data_len) {
     return ret;
 }
 
+static int connTLSWritev(connection *conn_, const struct iovec *iov, int iovcnt) {
+    if (iovcnt == 1) return connTLSWrite(conn_, iov[0].iov_base, iov[0].iov_len);
+
+    /* Accumulate the amount of bytes of each buffer and check if it exceeds NET_MAX_WRITES_PER_EVENT. */
+    size_t cum = 0;
+    for (int i = 0; i < iovcnt; i++) {
+        cum += iov[i].iov_len;
+        if (cum > NET_MAX_WRITES_PER_EVENT) break;
+    }
+
+    /* The amount of all buffers is greater than NET_MAX_WRITES_PER_EVENT, 
+     * which is not worth doing so much memory copying to reduce system calls,
+     * therefore, invoke connTLSWrite() multiple times to avoid memory copies. */
+    if (cum > NET_MAX_WRITES_PER_EVENT) {
```

Comment:
We're currently using the `SSL_MODE_ENABLE_PARTIAL_WRITE` and `SSL_MODE_ACCEPT_MOVING_WRITE_BUFFER` flags with OpenSSL to preserve `write()` compatibility, so I'm not sure there's real value in doing any kind of aggregation that OpenSSL would do itself.

Reply:
@yossigo so you suggest to chuck everything after that `if`, and let `connTLSWritev` just do repeated calls to `connTLSWrite`?

do you think there's a point to try to benchmark this?

Reply:
@oranagra Yes, I don't think we should add this mechanism without benchmarking and proving that there's real benefit in doing that. In theory it could be even counter productive, and it's possible that we'd get better results by tuning OpenSSL or using in it a different way (e.g. avoid those two mentioned flags when we have unmovable buffers and don't need `write()` compatibility).